### PR TITLE
Update service and delegate for AB#9734

### PR DESCRIPTION
**Pull request by sslaws**
# Fixes or Implements [AB#9734](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9734)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description

Updates the Immunization delegate, service and model to support the new data attributes.
Cleaned up a few smells and removed a couple of pragma's.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

* [X] Unit Tests Updated
* [X] Functional Tests Updated
* [ ] Not Required

Unit Tests were validated and functional tests updated.

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

No

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/Immunization/src/Models/ImmunizationModel.cs [hunk 1]
```diff
@@ -54,26 +65,28 @@ public static ImmunizationModel FromPHSAModel(ImmunizationResponse model)
                 Id = model.SourceSystemId,
                 Name = model.Name,
                 DateOfImmunization = model.OccurrenceDateTime,
+                ProviderOrClinic = model.ProviderOrClinic,
+                ImmunizationAgents = model.ImmunizationAgents,
```
Comment (leobcgov) [inline Apps/Immunization/src/Models/ImmunizationModel.cs:69]: Do we need to set the Status of the ImmunizationModel here?

**Review by leobcgov — CHANGES_REQUESTED** [review]

**Review by tiagograf-AOT — APPROVED** [review]

File: Apps/Immunization/src/Models/ImmunizationModel.cs (hunk 1 above)
Comment (leobcgov) [reply Apps/Immunization/src/Models/ImmunizationModel.cs:69]: Ping @sslaws in case you have missed this :-)

Comment (leobcgov) [reply Apps/Immunization/src/Models/ImmunizationModel.cs:69]: 9723 mentions "Don't include Due Date and Status for now" so I approve this PR

**Review by leobcgov — APPROVED** [review]

File: Apps/Immunization/src/Delegates/RestImmunizationDelegate.cs [hunk 2]
```diff
@@ -102,15 +103,13 @@ public async Task<RequestResult<IEnumerable<ImmunizationResponse>>> GetImmunizat
...
-                            PHSAResult<ImmunizationResponse>? phsaResult= JsonSerializer.Deserialize<PHSAResult<ImmunizationResponse>>(payload, options);
+                            PHSAResult<ImmunizationResponse>? phsaResult = JsonSerializer.Deserialize<PHSAResult<ImmunizationResponse>>(payload, options);
                             if (phsaResult != null && phsaResult.Result != null)
                             {
                                 retVal.ResultStatus = Common.Constants.ResultType.Success;
                                 retVal.ResourcePayload = phsaResult.Result;
                                 retVal.TotalResultCount = phsaResult.Result.Count;
-#pragma warning disable CA1305 // Specify IFormatProvider
-                                retVal.PageSize = int.Parse(this.immunizationConfig.FetchSize);
-#pragma warning restore CA1305 // Specify IFormatProvider
```
Comment (FuriousLlama) [inline Apps/Immunization/src/Delegates/RestImmunizationDelegate.cs:113]: +1!!!

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**